frame/staking/src/testing_utils.rs

```diff
@@ -34,6 +34,49 @@ use sp_std::prelude::*;
 
 const SEED: u32 = 0;
 
+// A helper enum to configure a payout destination without knowing the stash and controller
+// accounts beforehand.
+#[derive(PartialEq, Copy, Clone)]
+pub enum PayoutDestinationOpt<AccountId> {
```

Comment:
I am noticing a potential code smell to create almost identical structs for achieving the same functionality. I would rather suggest to create a `PayoutDestinationBuilder` that could take parameters such as controller, stash, RewardDestination and a PayoutSplitOpt to construct a `PayoutDestination`. This builder should be able to replace `RewardDestination::to_payout_destination` as well.